libs/header/sync/ranges.go

```diff
@@ -83,6 +98,26 @@ func (rs *ranges[H]) FirstRangeWithin(start, end uint64) (*headerRange[H], bool)
 	return nil, false
 }
 
+// FindAllRangesWithin returns all headerRanges that apply the requested range.
+// Returned ranges contain `start` within their bounds.
+// NOTE: ranges do not need to contain `end`.
```

Comment:
Can you give an example in this function's documentation of a case without an end?

Reply:
n/a